edge-util/src/Microsoft.Azure.Devices.Edge.Util/PersistentStorageValidation.cs

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Util
+{
+    using System.IO;
+    using Microsoft.Extensions.Logging;
+    using Newtonsoft.Json;
+
+    public static class PersistentStorageValidation
+    {
+        public static void ValidateStorageIdentity(string storagePath, string deviceId, string iotHubHostname, string moduleId, Option<string> moduleGenerationId, ILogger logger)
+        {
+            DeviceIdentity currentIdentity = new DeviceIdentity(deviceId, iotHubHostname, moduleId, moduleGenerationId.OrDefault());

```

Comment:
nit: you can use OptionConverter and set ModuleGenerationId as Option<string> in DeviceIdentity (example in CombinedDockerConfig)